inbm/dispatcher-agent/dispatcher/ota_parser.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, repo_type: str) -> None:
         self._username = None
         self._password = None
         self._signature = None
-        self._hash_algorithm: Optional[int] = None
+        self._hash_algorithm: int = DEFAULT_HASH_ALGORITHM
```

Comment:
**Suggestion:** Instead of setting 'self._hash_algorithm' to 'DEFAULT_HASH_ALGORITHM' directly in the constructor, consider setting it to 'None' and then ensure it is set to a default value in the 'parse' method if no specific 'sigversion' is provided. This maintains consistency with the other optional attributes and allows for the possibility that different parsing subclasses might have different default hash algorithms. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        self._hash_algorithm: Optional[int] = None
```

Reply:
this is just a revert of my change; I want it to be set to an int by default